File: task-03/Python/SubTask_4.py
import math

input=open("input.txt",'r')
output=open("output.txt",'w')

def printStars(n):
    for i in range(1,n+1):
        for y in range(1,n-i+1):
            output.write(" ")
        for x in range(1,2*i):
            output.write("*")
        output.write("\n")
    for i in range(n-1,0,-1):
        for y in range(1,n-i+1):
            output.write(" ",)
        for x in range(1,2*i):
            output.write("*")
        output.write("\n")

n=int(math.ceil(int(input.read())/2))
printStars(n)

File: task-03/Rust/SubTask_4.rs
use std::fs::File;
use std::io::{self,Read,Write,OpenOptions};
use std::f64;

fn main()->io::Result<()> {
    let mut input=File::open("input.txt")?;
    let mut data=String::new();
    input.read_to_string(&mut data)?;
    let n:usize=(data.trim().parse::<f64>().unwrap()/2.0).ceil() as usize;
    let mut output=OpenOptions::new()
        .write(true)
        .append(true)
        .open("output.txt")?;
    for i in 1..=n {
        for _ in 1..=n-i {
            write!(output," ")?;
        }
        for _ in 1..=2*i-1 {
            write!(output,"*")?;
        }
        writeln!(output)?;
    }
    
    for i in (1..n).rev() {
        for _ in 1..=n-i {
            write!(output," ")?;
        }
        for _ in 1..=2*i-1 {
            write!(output,"*")?;
        }
        writeln!(output)?;
    }
    Ok(())
}
